lib/src/utilities/snappable_scrollable_sheet.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class SnappableScrollableSheet extends StatefulWidget {
```

Comment:
@ueman So I sort of got the bottom sheet working with snapping (eg if the user drags past the midpoint, it snaps to the top. If they drag below the midpoint, it snaps to the bottom).

However, this is a janky mess because the `DraggableScrollableSheet` implementation is bad and should feel bad. Namely, it doesn't support snapping to the top or bottom (or any programmatic controls for that matter) so you have to do some gross hacks to get it to work. See: https://github.com/flutter/flutter/issues/34111, https://github.com/flutter/flutter/issues/45009

Do you mind if I use an external package such as [sliding_up_panel](https://pub.dev/packages/sliding_up_panel)? I think that's by far the best solution here.